File: Data Structures & Algorithms/minimum-difference-between-highest-and-lowest-of-k-scores/submission-0.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()

        l, r = 0, k - 1

        res = float("inf")
        
        while r < len(nums):
            diff = nums[r] - nums[l]
            res = min(res, diff)
            l, r = l + 1, r + 1

        return res

File: Data Structures & Algorithms/longest-continuous-subarray-with-absolute-diff-less-than-or-equal-to-limit/submission-0.java
class Solution {
    public int longestSubarray(int[] nums, int limit) {
        int res = 1;
        for (int i = 0; i < nums.length; i++){
            int mini = nums[i], maxi = nums[i];
            for (int j = i + 1; j < nums.length; j++) {
                mini = Math.min(mini, nums[j]);
                maxi = Math.max(maxi, nums[j]);

                if ((maxi - mini) > limit){
                    break;
                }
                res = Math.max(res, j - i + 1);
            }
        }

        return res;
    }
}